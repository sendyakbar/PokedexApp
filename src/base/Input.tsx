import React from 'react';
import {StyleSheet, View, TextInput, ViewStyle, TextStyle} from 'react-native';

type Props = {
  placeholder: string;
  onChangeText?: (input: string) => void;
  value?: string;
};

export default function Input(props: Props): JSX.Element {
  const {placeholder, onChangeText, value} = props;

  return (
    <View style={styles.container}>
      <TextInput
        placeholder={placeholder}
        onChangeText={onChangeText}
        value={value}
        style={styles.textInput}
        autoCapitalize="none"
      />
    </View>
  );
}

type StyleType = {
  container: ViewStyle;
  textInput: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    paddingVertical: 4,
    paddingHorizontal: 8,
    borderWidth: 1,
    borderColor: 'grey',
    borderRadius: 4,
  },
  textInput: {
    fontSize: 12,
    color: 'grey',
    padding: 0,
  },
});
