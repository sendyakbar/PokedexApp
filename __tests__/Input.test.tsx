import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import Input from '../src/base/Input';

it('renders correctly', () => {
  const tree = renderer
    .create(<Input placeholder="Test" onChangeText={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
