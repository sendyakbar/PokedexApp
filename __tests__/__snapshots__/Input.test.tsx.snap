// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "grey",
      "borderRadius": 4,
      "borderWidth": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Test"
    style={
      {
        "color": "grey",
        "fontSize": 12,
        "padding": 0,
      }
    }
  />
</View>
`;
